extensions/amp-image-slider/0.0/amp-image-slider.js

```diff
@@ -0,0 +1,566 @@
+    barHintIcon.classList.add('i-amphtml-image-slider-hint-icon');
+    this.barHint_.appendChild(barHintIcon);
+
+    this.barStick_.appendChild(this.barHint_);
+  }
+
+  /**
+   * Update element positions based on percentage
+   * @param {number} leftPercentage
+   */
+  updatePositions(leftPercentage) {
+    this.updateTranslateX(this.bar_, leftPercentage);
+
+    this.updateTranslateX(this.leftMask_, leftPercentage - 1);
+    this.updateTranslateX(this.leftAmpImage_, 1 - leftPercentage);
+    if (this.leftLabelWrapper_) {
+      this.updateTranslateX(this.leftLabelWrapper_, 1 - leftPercentage);
+    }
+  }
+
+  /**
+   * Set translateX of the element
+   * @param {Element} element
+   * @param {number} percentage
+   */
+  updateTranslateX(element, percentage) {
+    setStyle(element, 'transform', `translateX(${percentage * 100}%)`);
+  }
+
+  /**
+   * Register event handlers
+   */
+  registerEvents() {
+    if (this.isEventRegistered_) { // avoid dup
+      return;
+    }
+
+    if (this.isHoverSlider_) {
+      this.element.addEventListener('mousemove', this.handleHideHint, true);
+      this.container_.addEventListener('mousemove', this.handleHover);
+    } else {
+      // Use container_ for drag operation instead
+      // element for click/tap operations
+      this.element.addEventListener('mousedown', this.handleHideHint, true);
+      this.element.addEventListener('touchstart', this.handleHideHint, true);
+      this.element.addEventListener('click', this.handleClickImage);
+      this.element.addEventListener('touchend', this.handleTapImage);
+      dev().assertElement(this.barButton_)
+          .addEventListener('mousedown', this.dragStart);
+      dev().assertElement(this.barButton_)
+          .addEventListener('touchstart', this.touchStart);
+    }
+
+    this.isEventRegistered_ = true;
+  }
+
+  /**
+   * Unregister event handlers
+   */
+  unregisterEvents() {
+    if (this.isHoverSlider_) {
+      this.element.removeEventListener('mousemove', this.handleHideHint, true);
+      this.container_.removeEventListener('mousemove', this.handleHover);
+    } else {
+      this.element.removeEventListener('mousedown', this.handleHideHint, true);
+      this.element.removeEventListener('touchstart', this.handleHideHint, true);
+      this.element.removeEventListener('click', this.handleClickImage);
+      this.element.removeEventListener('touchend', this.handleTapImage);
+      dev().assertElement(this.barButton_)
+          .removeEventListener('mousedown', this.dragStart);
+      dev().assertElement(this.barButton_)
+          .removeEventListener('touchstart', this.touchStart);
+      // remove pointer related events below
+      this.dragEnd(null);
+      this.touchEnd(null);
+    }
+
+    this.isEventRegistered_ = false;
+  }
+
+  /**
+   * Handle hover event
+   * @param {Event} e
+   */
+  handleHover(e) {
+    // This offsetWidth may change if user resize window
+    // Thus not cached
```

Comment:
Oh okay, yes this is correct, but you can always update this value when the user resizes the window. This will be a lot less expensive than calculating it every time you `handleHover` or animate anything. You should update it in the override for `onMeasureChanged`.